shell/server/core.js

```diff
@@ -198,7 +198,11 @@ class PersistentUiViewImpl {
         throw new Error("grain no longer exists");
       }
 
-      const pkg = Packages.findOne({ _id: grain.packageId });
+      let pkg = Packages.findOne({ _id: grain.packageId });
+      if (!pkg) {
+        pkg = DevPackages.findOne({ appId: grain.appId });
```

Comment:
This can still fail to find a package, e.g. if the grain was created in dev mode but then added to a collection while `spk dev` was no longer running. So I suppose we need to add a `|| {}` here.